__tests__/app.test.js

```diff
@@ -164,5 +159,253 @@ describe("post/api/articles/:articleid/comments", () => {
+            expect.objectContaining({
+              username: expect.any(String),
+              name: expect.any(String),
+              avatar_url: expect.any(String),
+            })
+          );
+        });
+      });
+  });
+});
+
+describe("api queries", () => {
+  test("returns correct topic to the topic query", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toHaveLength(11);
+        body.articles.forEach((article) => {
+          expect(article).toEqual(
+            expect.objectContaining({
+              title: expect.any(String),
+              topic: expect.any(String),
+              created_at: expect.any(String),
+              article_img_url: expect.any(String),
+              article_id: expect.any(Number),
+              votes: expect.any(Number),
+              author: expect.any(String),
+              comment_count: expect.any(String),
+            })
+          );
+        });
+      });
+  });
 
+  test("returns correct order when using sort_by with order", () => {
+    return request(app)
+      .get("/api/articles?sort_by=topic&order=asc")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toBeSortedBy("topic", { ascending: true });
+      });
+  });
+  test("combines all 3 queries", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch&sort_by=author&order=asc")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toBeSortedBy("author", { ascending: true });
+      });
+  });
+  test("invalid query given returns data with default queries", () => {
+    return request(app)
+      .get("/api/articles?abcd")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toBeSortedBy("created_at", { descending: true });
+      });
+  });
+  test("invalid sort_by parameters cause error", () => {
+    return request(app)
+      .get("/api/articles?sort_by=break")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("invalid sort_by query");
+      });
+  });
+  test("invalid order parameters cause error", () => {
+    return request(app)
+      .get("/api/articles?order=break")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("invalid order query");
+      });
+  });
+  test("invalid order parameters cause error", () => {
+    return request(app)
+      .get("/api/articles?order=break")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("invalid order query");
+      });
+  });
+  test("topic parameters for topic that does not exist causes 404 error", () => {
+    return request(app)
+      .get("/api/articles?topic=break")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("No data for this topic");
+      });
+  });
+});
+
+describe("get/api/articles/:articleid/comments?queries", () => {
```

Comment:
11: Comment count isn't supposed to be a query! We're looking for you to retrospectively add comment_count as a key on your article body.